src/elevator/Elevator.java

```diff
@@ -73,17 +84,36 @@ public synchronized void processMessage() {
 			}
 		}
 		System.out.println("Elevator: Processing message in elevator...");
-		System.out.println("Elevator: " + msg.getBody());
+		loadFloorMessage(msg);
+		System.out.println("Elevator " + elevatorNumber +  ": " + msg.getBody());
+		
+		while(currFloor != currDest) {
+			if(currFloor < currDest) {
+				state.onNext(Transition.RECEIVEDMESSAGE_UP);
+				currFloor++;
+			} else if (currFloor > currDest) {
+				state.onNext(Transition.RECEIVEDMESSAGE_DOWN);
+				currFloor--;
+			} else {
+				state.onNext(Transition.REACHEDDESTINATION);
```

Comment:
It will never reach this state because when currFloor == currDest - it will leave the loop